test/external-modules/latency-simulating-directory/src/main/java/org/elasticsearch/test/inputlatency/InputLatencyPlugin.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.inputlatency;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexModule;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.plugins.Plugin;
+
+import java.util.List;
+
+/**
+ * Supplies a directory wrapper that simulates networks latencies when loading lucene
+ * directory files from disk
+ */
+public class InputLatencyPlugin extends Plugin {
+
+    private static final Logger log = LogManager.getLogger(InputLatencyPlugin.class);
+
+    /** Setting for enabling or disabling simulated object store latencies. Defaults to false. */
+    public static final Setting<Boolean> INPUT_LATENCY_ENABLED = Setting.boolSetting(
```

Comment:
nit: could it be a time value, and if > 0ms then directory will be wrapped and reads will be delayed?